### ISSUE TITLE
SPARQL Endpoint visualization not working

### ISSUE THREAD

**Issue by jakubklimek** [linkedpipes/visualizations#43]
Try http://statistics.gov.scot/sparql

When I click Visualize, it returns to homescreen

**Comment by jirihelmich** [comment]
`{"message":"ERROR: Endpoint returned Content-Type: text/html which is not currently supported for ASK queries"}`

**Comment by jakubklimek** [comment]
Hmm... so... a better error message and check whats wrong?

**Comment by jirihelmich** [comment]
Yep, working on it. But it seems that the URL you provide is not an endpoint, just some interface to it.

**Comment by jirihelmich** [comment]
```
<input type="hidden" name="authenticity_token" value="ymUL2+ehXIqznkE1Sr/jUbLNoDlpXDjLOkGNfhH6Iq9FTxjQWQEMCcLBzQ3e6tn0sSVbtBL8L3hBfgNpKLsUDw==">
```

**Comment by jakubklimek** [comment]
The API is described there in the API tab, so we should be able to use it.
What content-type do we request?

**Comment by jirihelmich** [comment]
```
Option 2: GET
Issue a GET to the following URL (note the .json extension - see the formats section for more detail on this):

ASK queries return a boolean result:

Format  Extensions  Accept Headers
XML .xml    application/xml,
application/sparql-results+xml
JSON    .json   application/json,
application/sparql-results+json
Text    .txt, .text text/plain
```

As the ASK query is short, Jena decides to fire a GET request. But to URL `http://statistics.gov.scot/sparql`, not `http://statistics.gov.scot/sparql.xml`.

That is highly unusual... it's not a standard SPARQL endpoint.

**Comment by jirihelmich** [comment]
Any thoughts for this?

**Comment by jakubklimek** [comment]
Well.. tried http://statistics.gov.scot/sparql.xml and it discovers DCV pipeline, which results in error that no dataset is found... which is weird.

So far, I would not do anything automatic with the endpoint if it will not display anything anyway

**Comment by jirihelmich** [comment]
We use all ask, construct and select and they have different extensions for
these. For construct, you need to query sparql.rdf instead of sparql.xml.
Thu, 10 Mar 2016 v 18:43 odesílatel Jakub Klímek notifications@github.com
napsal:
